src/routes/quickstarts/react-native/react-native-testing-library.mdx

```diff
@@ -0,0 +1,242 @@
+# Mock Network Requests in React Native Testing Library with Mirage
+
+Use your Mirage server to test your React Native application under different server scenarios using [React Native Testing Library](https://github.com/testing-library/native-testing-library) and [Jest Native](https://github.com/testing-library/jest-native).
+
+> This is a quickstart guide should also work for people using [`react-test-renderer`](https://reactjs.org/docs/test-renderer.html) or [`react-native-testing-library`](https://github.com/callstack/react-native-testing-library).
+
+## Step 1: Install Mirage
+
+First, make sure you have Mirage installed:
+
+```bash
+# Using npm
+npm install --save-dev miragejs
+
+# Using Yarn
+yarn add --dev miragejs
+```
+
+## Step 2: Setup your testing environment
+
+Out of the box, React Native does a good job to polyfill browser API's that we might need in libraries that were originally built to run on the browser. Our problem when using Mirage in a Jest testing environment is that we don't have access to some properties that the browser provides by default. To get around this issue we must update Jest's environment immediately after its setup is finished.
+
+To update your Jest environment follow these steps:
+
+1. Create a setup file in the root of your project. For the purpose of this guide we'll call it `jest.setup.js`.
+
+2. Install a polyfill for XMLHttpRequest:
+
+```bash
+# Using npm
+npm install --save-dev xmlhttprequest
+
+# Using Yarn
+yarn add --dev xmlhttprequest
+```
+
+3. Patch the variable `global` in your `jest.setup.js` to add support for XMLHttpRequest, self and window.
+
+```js
+// jest.setup.js
+global.self = global;
+global.window = {};
+global.XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+```
+
+4. Tell Jest to load the environment that we've created by adding the [`setupFilesAfterEnv`](https://jestjs.io/docs/en/configuration.html#setupfilesafterenv-array) to your Jest configuration. We must add `jest.setup.js` as an item in the list of files that Jest must execute after setting up its environment.
+
+> Your jest configuration can be found on your package.json or in the root of your project with the name `jest.config.js`.
+
+```js
+// jest.config.js
+const jestPreset = require('@testing-library/react-native/jest-preset');
+
+module.exports = {
+  preset: '@testing-library/react-native',
+  setupFiles: [...jestPreset.setupFiles],
+  setupFilesAfterEnv: ['./jest.setup.js'],
+};
+```
+
+## Step 3: Define your server
+
+Create a new `server.js` file and define your mock server.
+
+Here's a basic example:
+
+```js
+// src/server.js
+import { Server, Model } from "miragejs"
+
+export function makeServer({ environment = "development", trackRequests = false } = {}) {
```

Comment:
Any specific reason for passing in `trackRequests`? I think we should follow the make server definition from the other quick start guides on the site.

However if this is something that's needed I'd love to know more!

Reply:
I added `trackRequests` here because I thought that I'd be good for people to know that Mirage lets you do that. Now that I think about it, adding `trackedRequests` and making assertions against the requests adds more complexity to the examples and is out of the scope of a quick guide.